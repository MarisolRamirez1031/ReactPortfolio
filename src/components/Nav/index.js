import React from "react";

function Nav() {
    const categories = [
        {
          name: "Portfolio",
          description:
            "Photos of grocery stores, food trucks, and other commercial projects",
        },
        { name: "Resume", description: "Portraits of people in my life" },
      ];

      function categorySelected(name) {
        console.log(`${name} clicked`)
      }

    return (
<header>
  <h2>
    <a data-testid="link" href="/">
      <span role="img" aria-label="sunflower"> ðŸŒ»</span> Marisol Ramirez
    </a>
  </h2>
  <nav>
    <ul className="flex-row">
      <li className="mx-2">
      <a data-testid="about" href="#about">
          About me
        </a>
      </li>
      <li>
        <span>Contact</span>
      </li>
       {categories.map((category) => (
        <li
          className="mx-1"
          key={category.name}
        >
          <span onClick={() => categorySelected(category.name)}>
            {category.name}
          </span>
        </li>
      ))}
    </ul>
  </nav>
</header>
    )
}

export default Nav;