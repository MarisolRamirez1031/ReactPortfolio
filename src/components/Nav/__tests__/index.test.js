import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from "..";

afterEach(cleanup);

describe('Nav component renders', () => {
  it('renders', () => {
    render(<Nav />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Nav />);
    
    expect(asFragment()).toMatchSnapshot();
  });
})

describe('emoji is visible', () => {
  it('inserts emoji into the h2', () => {
  const { getByLabelText } = render(<Nav />);

  expect(getByLabelText('sunflower')).toHaveTextContent('ðŸŒ»');
  });
})  

describe('links are visible', () => {
  it('inserts text into the links', () => {
    const { getByTestId } = render(<Nav />);

    expect(getByTestId('link')).toHaveTextContent('Marisol Ramirez');
    expect(getByTestId('about')).toHaveTextContent('About me');
  });

})