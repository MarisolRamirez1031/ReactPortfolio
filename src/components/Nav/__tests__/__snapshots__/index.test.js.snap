// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="sunflower"
          role="img"
        >
           ðŸŒ»
        </span>
         Marisol Ramirez
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Portfolio
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Resume
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
